### ISSUE TITLE
Request: Migrate DynamoDbPersistenceAdapter to AWS SDK v3 and save 60Mb

### ISSUE THREAD

**Issue by talkingnews** [alexa/alexa-skills-kit-sdk-for-nodejs#696]
Currently,  DynamoDbPersistenceAdapter has a peer dependency of aws-sdk, which install 65Mb of stuff we don't need.

Whereas the current slimline v3 version of @aws-sdk/client-dynamodb is 4.97 Mb including dependencies.
https://www.npmjs.com/package/@aws-sdk/client-dynamodb

Is anyone working on a re-write, as this library is currently the difference between my skill package being 7Mb or 70Mb!

**Comment by sjcomstock67** [comment]
agree this is a pain point. Will try to bump it up on the priority list.

**Comment by sam-goodwin** [comment]
We definitely want to get to this. @talkingnews just so I understand, are you using any type of bundling/tree-shaking like webpack or esbuild for your skill? This might be especially helpful if you are running in AWS Lambda since the SDKs are provided by the container - bundle the skill and mark `aws-sdk` as external - the bundler will not include any AWS SDK code in your bundle, leaving you with a tiny JS file you can run in your Lamdba.

I understand this might not always be the right solution, so we should still create a new version of the adapter which doesn't depend on the whole AWS SDK.

**Comment by sjcomstock67** [comment]
@doiron this was started by @sam-goodwin and there's a linked PR. We just need to finish it up. Will discuss during sprint planning.

**Comment by rahulawl** [comment]
Is this issue/feature-request still relevant?
We are working on prioritization of relevant issues and cleanup of rest. If we don’t hear back in 2 weeks, we will assume that the issue is not relevant and we will close it.

**Comment by seanfisher** [comment]
@rahulawl Yes, still relevant. See [my comments](https://github.com/alexa/alexa-skills-kit-sdk-for-nodejs/issues/697#issuecomment-1378217839) in #697.

**Comment by crackmac** [comment]
@rahulawl Yes, still relevant.

**Comment by tcutts** [comment]
This is particularly relevant now with AWS v2 API being withdrawn in September 2025.  Everyone who's written an Alexa skill using ASK needs ASK to support the v3 API well in advance of that deadline.
